### ISSUE TITLE
before_all remove resource from DB after first `each` hook

### ISSUE THREAD

**Issue by zauzaj** [palkan/test-prof#107]
Hello,

We started using `test-prof` library to speed up our current specs, but we struggled into very weird issue.
So, the current specs are looking as bellow:

```
  before_all do
    @calendar_access = create(:calendar_access)
    @agenda = @calendar_access.agenda
  end

  let(:agenda) { @agenda }
  let(:calendar_access) { @calendar_access }
  let(:start) { 1.day.ago }

  before(:each) do
    binding.pry
    @diff = ConsultationsDiff.new(agenda, old_consultations, old_consultations, new_consultations)
  end

..........
```

So first example passes well, but on second one in debugger we're over:

```
agenda.reload #=> Couldn't find DoctorAgenda with 'id'=145 [WHERE "agendas"."type" IN ('DoctorAgenda')]
@agenda.reload #=> Couldn't find DoctorAgenda with 'id'=145 [WHERE "agendas"."type" IN ('DoctorAgenda')]
Agenda.all #=> []
....
same for `callendar_access`. 

I think it's because of DatabaseCleaner but when I comment it the issue remains. 

Do you have any thought on this ?

Any help is appreciated.

Regards

**Comment by palkan** [comment]
Hi @zauzaj,

Did you solve the issue?

**Comment by zauzaj** [comment]
@palkan Hi, sorry for late response. Actually the main problem was old gem version so after we updated it everything works as normal. 

I just checked your note here https://github.com/palkan/test-prof/issues/108, and I think you actually right. So we don't need to use DatabaseCleaner at all as we have `active_record_shared_connection`. 
I'm updating our specs to `test-prof` so I will definitely investigate for those scenarios.

Will keep you posted ;)
